### PR TITLE
Mining improvements and changes- Pt .2: Generation and Mobs

### PR DESCRIPTION
**Pull request by Sun-Soaked**
Yes I know I finished part 2 before part 1 was even fully merged, it just be like that sometimes.

## About The Pull Request
- Adds gemstones! You can buy stuff by selling them, or hoard them like a dragon. More functionality pending, maybe.
- There's a [BIG SHOT] in all of us, if you can find the mask of riches...
- The necropolis is stirring, and the tendrils it punches through earth have grown far more formidable. 
- A strange demonic incursion can now be found across the surface of ice worlds- the rewards for destroying these structures can be quite considerable, but beware- that which exits may not be happy to see you. The new portals have a slight tendency to pull objects across time and space.
- Adds and nerfs both ice whelps and ice demons.
- Asteroid spawners now drop ore, as well as occasional scraps of mining paraphernalia.
- Adds crusher trophies to the polar bear and white wolf. 
- Puce now grows everywhere, and fireblossoms now grow on lavaland.
- Boosts mob elite spawn chances slightly, and considerably boosts drop chance for elite trophies. These used to be nearly unobtainable, but can now generally be acquired in-round.
- Crusher users will rejoice, for ancient goliaths and dwarf legion now have crusher trophies. The trophy for the normal goliath now works as intended. 
- Adds three new elites: The Polar Warbear, Alpha Wolf, and Primordial demon. ...enjoy.

## Why It's Good For The Game
Generally amplified the rewards and compounded the danger of both spawner types, alongside my other mining PR- this supports our current setup of unlimited possible rewards, and is setting up for a world where smashing tendrils and mugging megafauna for their lunch money is a team effort.
Asteroid good
Other assorted mining gen improvements.

## Changelog
:cl:
add: Adds a wolf talisman that can be crafted from both wolf trophies.
add: The legion staff, allowing one to summon legion skulls to their side.
tweak: makes legion and wolf trophies drop independent of crusher usage.
balance: Nerfs money gained from selling necropolis loot.
add: gemstones! They can be sold on export for money, or held onto I guess. Adds rare chances for them to drop to a few mobs, and to a particular demonic pool.
add: A particularly BIG mask that alters speech like the Italian moustache.
balance: Reduces the health and functional speed/range of whelps and ice demons.
balance: increases the health and maximum spawns of lava tendrils.
tweak: Makes asteroid spawners drop ore as well, with a small chance of other loot.
add: implements demonic portals on snow planets. 
add: Ports the white wolf and polar bear crusher trophies.
tweak: significantly enriched demonic portal loot drops and added scaling difficulty- You are not safe when it dies, so be ready to fight off whatever comes through.
tweak: adds on-contact hostile versions of clown mobs.
tweak: non-clowns can now wear the Cosmohonk hardsuit.
tweak: adds puce to all 3 high-difficulty planets. Adds fireblossom to the lavaland flora pool.
tweak: Increases elite spawnchance slightly, and significantly increases chances for elites to drop their unique crusher trophy
add: A new polar bear elite, a wolf elite, a demon elite and a trophy alongside each. 
tweak: Adds unique trophies for the ancient goliath, dwarf legion, and ash drake(old ash drake trophy is now the ice whelp trophy)
fix: goliath tentacle crusher trophy now works as intended, instead of doing nothing.
/:cl:

### CODE REVIEW

**Comment by Sun-Soaked** [comment]
ignore the monke stuff, I fucked up my base fork and had to fix it

File: code/game/objects/structures/icemoon/cave_entrance.dm [hunk 1]
```diff
@@ -133,17 +137,21 @@ GLOBAL_LIST_INIT(ore_probability, list(
...
 			new /obj/item/reagent_containers/glass/bottle/potion/flight(loc)
 		if(4)
 			new /obj/item/organ/heart/cursed/wizard(loc)
 		if(5)
-			new /obj/item/jacobs_ladder(loc)
+			new /obj/item/clothing/suit/space/hardsuit/quixote(loc)
+			new /mob/living/simple_animal/hostile/netherworld/blankbody(loc)
 		if(6)
 			new /obj/item/rod_of_asclepius(loc)
+			new /obj.effect/mob_spawn/human/doctor(loc)
```
Comment (ZephyrTFA) [inline code/game/objects/structures/icemoon/cave_entrance.dm]:
```suggestion
			new /obj/effect/mob_spawn/human/doctor(loc)
```

File: code/game/objects/structures/icemoon/cave_entrance.dm [hunk 2]
```diff
@@ -152,41 +160,93 @@ GLOBAL_LIST_INIT(ore_probability, list(
...
+			new /mob/living/simple_animal/hostile/bear/snow(loc)
+			new /obj/effect/decal/remains/human(loc)
 		if(27)
 			new /obj/item/book/granter/spell/sacredflame(loc)
+			new /obj/effect/temp_visual/fire(loc)
 		if(28)
 			new /mob/living/simple_animal/hostile/megafauna/blood_drunk_miner/doom(loc)
+		if(29)
+			new /mob/living/simple_animal/hostile/megafauna/swarmer_swarm_beacon(loc)//god help you
+		if(30)
```
Comment (ZephyrTFA) [inline code/game/objects/structures/icemoon/cave_entrance.dm]: All of these right here are a hard no

File: code/game/objects/structures/lavaland/necropolis_tendril.dm [hunk 3]
```diff
@@ -7,8 +7,8 @@
 	icon_state = "tendril"
 
 	faction = list("mining")
-	max_mobs = 3
-	max_integrity = 250
+	max_mobs = 5//shiptest edit, boosted power and health
```
Comment (ZephyrTFA) [inline code/game/objects/structures/lavaland/necropolis_tendril.dm]: Comment isn't needed

**Review by ZephyrTFA — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/icemoon/cave_entrance.dm (hunk 2 above)
Comment (Sun-Soaked) [reply code/game/objects/structures/icemoon/cave_entrance.dm]: The swarmer beacon now drops a small stack of swarmers and an infiltrator engi, and the syndicate now has if/either logic for either reward

**Comment by Sun-Soaked** [comment]
WAAAAA

**Comment by Sun-Soaked** [comment]
this WILL become a merge conflict when the mining rework PR is merged, I'm almost sure of it

File: code/modules/mob/living/simple_animal/hostile/clown.dm [hunk 4]
```diff
@@ -310,3 +310,27 @@
 	melee_damage_upper = 20
 	armour_penetration = 30
 	environment_smash = ENVIRONMENT_SMASH_WALLS//no escape
+
+/mob/living/simple_animal/hostile/clown/stacked
```
Comment (Anderson-Cooper) [inline code/modules/mob/living/simple_animal/hostile/clown.dm:314]: i opened this at school and now i am frightened. I want my mom to pick me up early.

Comment (Sun-Soaked) [reply code/modules/mob/living/simple_animal/hostile/clown.dm:314]: join in the fun!join in the fun!join in the fun!

Comment (Anderson-Cooper) [reply code/modules/mob/living/simple_animal/hostile/clown.dm:314]: stop, i am filled with fear

**Comment by HighPrincessErinys** [comment]
Cursed Buisnessman's Mask is in itself a typo but I must also question why it's [KRONOR] and not [KROMER]

**Comment by triplezeta** [comment]
probably to make it slightly less on the nose

**Comment by HighPrincessErinys** [comment]
also, consider nerfing the ice whelps further, probably by lowering their action speed (if that makes sense) and or their health. 300 health on a thing that can give you massive firestacks and acts slightly faster than a goliath is, while also doing a flat 20 damage per hit when a goliath does 12 to 20 damage per hit randomly is. nah.

**Comment by Sun-Soaked** [comment]
they *are* considerably slower than they used to be, already- but it's important to note that their skin gives a guaranteed path to making one of the strongest armors in the game(this is just how they already were) so I don't want to push them too low. They're not quite as ruthless now, and their fire attacks happen less frequently, and they spawn less often- but further than that, I'm not sure about pushing them lower

**Comment by HighPrincessErinys** [comment]
at the very (extra) least give their attacks some rng, even if its more high end than goliaths. say, 16-20 just for a little variety in your mauling experience

File: code/modules/cargo/exports/lavaland.dm [hunk 5]
```diff
@@ -72,3 +107,34 @@
 	cost = 10000
 	unit_name = "H.E.C.K. helmet"
 	export_types = list(/obj/item/clothing/head/helmet/space/hostile_environment)
+
+//not technically lavaland but this had a useful infrastructure to store them under
+/datum/export/lavaland/gems/diamond
```
Comment (TheNeoGamer42) [inline code/modules/cargo/exports/lavaland.dm]: uuuuuuuuuuhhhhhhhhhhhhhhhh this is the same thing as the thing below it which will probably break shit?

Comment (Sun-Soaked) [reply code/modules/cargo/exports/lavaland.dm]: fixed

**Pull request merged**